store-on-wpcom/assets/css/admin/nav-unification.css

```diff
@@ -0,0 +1,4 @@
+/* Fix header overflow https://github.com/woocommerce/woocommerce-admin/issues/7035 */
+.jetpack-connected .woocommerce-layout__header {
```

Comment:
`.jetpack-connected` rule is used to override the `.woocommerce-layout__header` which style is enqueued after ours. The assumption is also correct, i.e: without jetpack connected, nav unification will not be enabled.